packages/core/test/integration/segment_maintained_across_shared_promise.test.js

```diff
@@ -70,3 +70,6 @@ function sendRequest(address, cb) {
     .on('error', cb)
     .end();
 }
+
+// Cancel the patch because it doesn't affect other tests
+require('../../lib/patchers/promise_p').uncapturePromise();
```

Comment:
So this is effectively cleanup for the integration test? If so, can we add it after the assertion so we're confident it's executed after the test will execute? Or perhaps wrap things in a `try..finally` block? Please inform me if I'm misunderstanding the purpose here.

Reply:
@willarmiros Thank you for your review. I moved cleanup for integration test after assertion, and change setup and cleanup process in `promise_p.test.js` too.